src/main/groovy/com/github/jk1/license/reader/ProjectReader.groovy

```diff
@@ -20,36 +18,44 @@ class ProjectReader {
     ProjectData read(Project project) {
         ProjectData data = new ProjectData()
         data.project = project
-        // Get the configurations matching the name: that's our base set
-        final Set<Configuration> toReport = new HashSet<Configuration>(project
-                .getConfigurations().matching(new Spec<Configuration>() {
-            @Override
-             boolean isSatisfiedBy(Configuration configuration) {
-                for (String configurationName : project.licenseReport.configurations) {
-                    if (configuration.getName().equalsIgnoreCase(configurationName)) {
-                        if (configuration.hasProperty("canBeResolved") && !configuration.canBeResolved) {
-                            throw new GradleException("Project Reader: " +
-                                    "The specified configuration \"${configuration.name}\" can't be resolved. " +
-                                    "Try specifying a more specific configuration by adding flavor(s) and/or build type.")
-                        }
-                        return true
-                    }
-                }
-                return false
-            }
-        }))
-
-        // Now, keep adding extensions until we don't change the set size
-        for (int previousRoundSize = 0; toReport.size() != previousRoundSize; previousRoundSize = toReport.size()) {
-            for (Configuration configuration : new ArrayList<Configuration>(toReport)) {
-                toReport.addAll(configuration.getExtendsFrom())
-            }
-        }
-        LOGGER.info("Configurations: " + toReport.join(','))
-        for (Configuration configuration : toReport) {
+
+        final Set<Configuration> configurationsToScan = findConfigurationsToScan(project)
+
+        configurationsToScan.addAll(getAllExtendedConfigurations(configurationsToScan))
+
+        LOGGER.info("Configurations: " + configurationsToScan.join(','))
+        for (Configuration configuration : configurationsToScan) {
             LOGGER.info("Reading configuration: " + configuration)
             data.configurations.add(configurationReader.read(project, configuration))
         }
         return data
     }
+
+    private static Set<Configuration> getAllExtendedConfigurations(Set<Configuration> configurationsToScan) {
+        configurationsToScan.collect { it.extendsFrom }.flatten()
+    }
+
+    private static Set<Configuration> findConfigurationsToScan(Project project) {
+        project.configurations.findAll { config ->
+            isConfigured(config, project) && isResolvable(config)
+        }
+    }
+
+    private static boolean isConfigured(Configuration config, Project project) {
+        def configurationsToScan = project.licenseReport.configurations ?: project.configurations.collect {
+            it.name
+        }
+        config.name in configurationsToScan
+    }
+
+    private static boolean isResolvable(Configuration config) {
```

Comment:
What about custom configurations? They can also be marked as not-resolvable, but this will break then.
Using `configuration.hasProperty("canBeResolved") && !configuration.canBeResolved` isn't so bad I guess.  Also this property is in the officially supported https://docs.gradle.org/current/javadoc/org/gradle/api/artifacts/Configuration.html#isCanBeResolved--

So using the dynamic way to check if a configuration is resolvable is more robust.

Reply:
I agree to using the recommended way. I was looking at that. Having to check for the existence of that property before accessing it, is not ideal IMO. But if that is recommended I will use it.

Reply:
I think the checking for the property-existence should not be needed at all. This property is there and if it's not longer there in future versions, the plugin should crash. Also I don't think that this is the *recommended* way. It's just A way...
So adding the more general check but without existence-check will be probably the best solution.

Reply:
That's what I tried in the first place - it threw property missing exceptions. That's why I went with the static check on those unresolvable configurations at first.